File: Faltantes o por revisar/BurstBalloonMaximizeCoins.py
# Python program burst balloon problem.  
  
def getMax(A): 
    N = len(A) 
    A = [1] + A + [1]# Add Bordering Balloons 
    dp = [[0 for x in range(N + 2)] for y in range(N + 2)]# Declare DP Array 
      
    for length in range(1, N + 1): 
        for left in range(1, N-length + 2): 
            right = left + length -1
  
            # For a sub-array from indices left, right 
            # This innermost loop finds the last balloon burst 
            for last in range(left, right + 1): 
                dp[left][right] = max(dp[left][right], \ 
                                      dp[left][last-1] + \ 
                                      A[left-1]*A[last]*A[right + 1] + \ 
                                      dp[last + 1][right]) 
    return(dp[1][N]) 
  
# Driver code 
A = [1, 2, 3, 4, 5] 
print(getMax(A)) 

File: Faltantes o por revisar/BurstBalloonMaximizeCoins.java
// Java program to illustrate 
// Burst balloon problem 

public class BurstBalloonMaximizeCoins {

    public static int getMax(int[] A, int N) {

        // Add Bordering Balloons 
        int[] B = new int[N + 2];
        B[0] = B[N + 1] = 1;

        for (int i = 1; i <= N; i++) {
            B[i] = A[i - 1];
        }

        // Declaring DP array 
        int[][] dp = new int[N + 2][N + 2];

        for (int length = 1;
                length < N + 1; length++) {
            for (int left = 1;
                    left < N - length + 2; left++) {
                int right = left + length - 1;

                // For a sub-array from indices 
                // left, right. This innermost 
                // loop finds the last balloon burst 
                for (int last = left;
                        last < right + 1; last++) {
                    dp[left][right] = Math.max(
                            dp[left][right],
                            dp[left][last - 1]
                            + B[left - 1] * B[last]
                            * B[right + 1]
                            + dp[last + 1][right]);
                }
            }
        }
        return dp[1][N];
    }

    public static void main(String args[]) {
        int[] A = {1, 2, 3, 4, 5};
        int N = A.length;
        System.out.println(getMax(A, N));
    }
}
